src/Simple_Meshing.cc

```diff
@@ -34,6 +36,395 @@
+std::vector< std::array<size_t, 3> > Tile_Contours(
+        std::reference_wrapper<contour_of_points<double>> A,
+        std::reference_wrapper<contour_of_points<double>> B ){
+
+    contour_of_points<double> contour_A = A.get();
+    contour_of_points<double> contour_B = B.get();
+
+    const auto N_A = contour_A.points.size();
+    const auto N_B = contour_B.points.size();
+    if( N_A == 0 ){
+        throw std::invalid_argument("Contour A contains no vertices. Cannot continue.");
+    }
+    if( N_B == 0 ){
+        throw std::invalid_argument("Contour B contains no vertices. Cannot continue.");
+    }
+
+    // Determine contour orientations. Single-vertex contours can take any orientation, so use a reasonable default.
+    auto ortho_unit_A = vec3<double>(0.0, 0.0, 1.0);
+    auto ortho_unit_B = vec3<double>(0.0, 0.0, 1.0);
+    try{
+        ortho_unit_A = contour_A.Estimate_Planar_Normal();
+    }catch(const std::exception &){}
+    try{
+        ortho_unit_B = contour_B.Estimate_Planar_Normal();
+    }catch(const std::exception &){}
+
+    // Ensure the contours have the same orientation.
+    if(ortho_unit_A.Dot(ortho_unit_B) <= 0.0){
+        // Handle special cases
+        if( (N_A == 1) && (N_B != 1) ){
+            ortho_unit_A = ortho_unit_B;
+        }else if( (N_A != 1) && (N_B == 1) ){
+            ortho_unit_B = ortho_unit_A;
+
+        // Otherwise, flip one of the contours, recurse, and adjust the face labels to point to the actual vertex
+        // layout.
+        //
+        // Note: This effectively ignores contour orientation altogether.
+        }else{
+            YLOGWARN("Ignoring adjacent contours with opposite orientations. Recursing..");
+            std::reverse( std::begin(contour_B.points), std::end(contour_B.points) );
+            auto faces = Tile_Contours(A, std::ref(contour_B) );
+
+            for(auto &face_arr: faces){
+                for(auto &v_i : face_arr){
+                    if(N_A <= v_i) v_i = N_A + (N_B - 1) - (v_i - N_A);
+                }
+            }
+            return faces;
+        }
+    }
+
+    /***************************************************************************
+     * Graph theory search
+     **************************************************************************/
+
+    std::vector<std::vector<node>> nodes;
+
+
+    // Keep track of where we are looping over points
+    auto iter_A = std::begin(contour_A.points);
+
+    // const convenience iterators
+    const auto begin_A = std::begin(contour_A.points);
+    const auto begin_B = std::begin(contour_B.points);
+    const auto end_A = std::end(contour_A.points);
+    const auto end_B = std::end(contour_B.points);
+
+    auto p_i_next = *iter_A;
+    iter_A = std::next(iter_A);
+
+    size_t m = N_A - 1;
+    size_t n = N_B - 1;
+
+    // We repeat over A twice, since to search the graph we need to search repeats
+    // Don't loop over all points since last point is repeat of first
```

Comment:
could we change this slightly by checking that the last point is a repeat of the first? according to Hal's implementation of closed contours, a closed contour can have the last point as the first point but it is not necessary. (I'm bringing this up because this affected my 2 to 1 branching after Dominic changed the behavior of CreateCustomContours)

Reply:
Oh I didn't know it didn't have to be a repeat. The fact that they repeat causes a lot of headache when handling indexing, but sure I can change it to check.